src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs

```diff
@@ -176,7 +176,7 @@ IEnumerable<string> BuildSearchPath(string s)
 			var xamlPaths = from item in sources.SelectMany(BuildSearchPath)
 							select Path.GetDirectoryName(item);
 
-			return xamlPaths.Distinct();
+			return xamlPaths.Distinct().Where(x => !string.IsNullOrEmpty(x));
```

Comment:
Looks more of a workaround? We shouldn't be getting null/empty path I think?

Reply:
Yep. If you can provide advice on how to tell why empty strings are appearing (& how to prevent them) I'll happily fix them there too.

Reply:
I don't, but wanted to make sure I understand the change. In case I got to make changes to this generator, I'll remember to check that and let you know

Reply:
Having done some digging - here's what's happening.
I have a library that references Uno.UI.RemoteControl but contains no XAML files. When the library is built, [the Source generators recognize this as a "ValidPlatform"](https://github.com/unoplatform/uno/blob/master/src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs#L49) and so adds a `XamlSearchPath` attribute to the library even though there are not XAML files.
Then, when the application that references the library is built, the source generators for the app find the (empty string) attribute and adds that to the list to pass to the RemoteControlServer to watch as part of XamlHotReload.

I've now added some extra checks to avoid the empty strings being added to attributes.
This seemed safer than trying to update the existing logic to handle my case for 2 reasons:
1 - With no test coverage for this part of the code, I wanted to do the minimum I could that would mitigate the chance of breaking something while addressing the bug.
2 - This avoids any other (potential, future) scenario that might cause empty strings from ending up in that list from causing problems.

Reply:
It's indeed appropriate to have a failsafe at that location.